via/request_tools/error_handling.py

```diff
@@ -0,0 +1,40 @@
+"""Helpers for capturing requests exceptions."""
+
+from functools import wraps
+
+from requests import RequestException, exceptions
+
+from via.exceptions import BadURL, UnhandledException, UpstreamServiceError
+
+REQUESTS_BAD_URL = (
+    exceptions.MissingSchema,
+    exceptions.InvalidSchema,
+    exceptions.InvalidURL,
+    exceptions.URLRequired,
+)
+REQUESTS_UPSTREAM_SERVICE = (
+    exceptions.ConnectionError,
+    exceptions.Timeout,
+    exceptions.TooManyRedirects,
+    exceptions.SSLError,
+)
+
+
+def handle_errors(inner):
+    """Translate errors into our application errors."""
+
+    @wraps(inner)
+    def deco(*args, **kwargs):
+        try:
+            return inner(*args, **kwargs)
+
+        except REQUESTS_BAD_URL as err:
+            raise BadURL(err.args[0]) from None
+
+        except REQUESTS_UPSTREAM_SERVICE as err:
+            raise UpstreamServiceError(err.args[0]) from None
+
+        except RequestException as err:
+            raise UnhandledException(err.args[0]) from None
```

Comment:
Is the `from None` existing behaviour here? That looks like it deliberately swallows the original error but I'm not sure why we'd want to do that? I'd expect `raise CustomErrorClass(...) from err`

Reply:
Also not sure about the `err.args[0]`'s. Are those really necessary / why do we do that? It looks like it might crash if `requests` every raises an exception where `err.args` is empty

Reply:
Yeah, looks like both of these came from the existing code. We might want to consider changing it to `raise CustomErrorClass() from err` (with no arguments to `CustomErrorClass`)

Reply:
Requests exceptions happen to always have a message, so it's actually fine. It would be potentially dodgy in general.

I'll change it over. It's purely for us when debugging. We don't rely on the exact message at all when presenting the errors in the interface (just the class).